### DISCUSSION TITLE
check_sidekiq_queue script returns ERR instead of integer

### DISCUSSION THREAD

**Discussion by Ogtay** [sidekiq/sidekiq#6275] [Q&A]
I am using script (https://github.com/wanelo/nagios-checks/blob/master/check_sidekiq_queue) to monitor sidekiq queue on redis server. On nagios side status returns "SIDEKIQ OK : redis ERR on default" whereas it is supposed to be number instead of ERR. When I check that script directly on the server see it is not able to identify ERR type:

ubuntu@redis:/usr/local/nagios/libexec$ sudo ./check_sidekiq_queue
./check_sidekiq_queue: line 141: [: ERR: integer expression expected
./check_sidekiq_queue: line 143: [: ERR: integer expression expected
SIDEKIQ OK :  redis ERR on default|sidekiq_queue_default=ERR;50;100

ubuntu@redis:/usr/local/nagios/libexec$ sudo ./check_sidekiq_queue  -q all
./check_sidekiq_queue: line 127: [: ERR: integer expression expected
./check_sidekiq_queue: line 127: [: ERR: integer expression expected
./check_sidekiq_queue: line 127: [: ERR: integer expression expected
SIDEKIQ OK : redis -1 on none|sidekiq_queue_none=-1;50;100

ubuntu@redis:/usr/local/nagios/libexec$ sudo ./check_sidekiq_queue  -q none
./check_sidekiq_queue: line 141: [: ERR: integer expression expected
./check_sidekiq_queue: line 143: [: ERR: integer expression expected
SIDEKIQ OK : redis ERR on none|sidekiq_queue_none=ERR;50;100

Directly checking with redis-cli command seems it returns correct value:

ubuntu@redis:~$ redis-cli -p 6379 -a XXXXXXXXXXXX -n 12 llen default
Warning: Using a password with '-a' or '-u' option on the command line interface may not be safe.
(integer) 0

Did you face same/similar issue before or any idea ?

**Comment by mperham** [comment]
I don't know anything about that script. Why don't you open an issue in that repo so the developer can help you?

**Reply by Ogtay** [reply]
@mperham I dont see discussions tab there, This repository has been archived by the owner on Jun 2, 2020. It is now read-only.

**Reply by mperham** [reply]
I'm not really strong on bash, there's obviously some issue that isn't obvious to me. The basic idea works for me:

```
% redis-cli llen queue:default 
(integer) 1
```

**Reply by Ogtay** [reply]
@mperham thank you for checking. redis-cli on the server works but script doesn't return expected value. Is there a way to re-open that repo and reach out to developers?

**Reply by mperham** [reply]
That would be rude. You need to understand that when you use open source software, you have no warranty and they are under no obligation to support you. By marking it read only, they are saying "We're done, you're on your own".
